### PR TITLE
Setting CSI GCP topology label on machine deployment creation to allo…

### PR DESCRIPTION
**Pull request by elankath**
…w scale-up

**How to categorize this PR?**

/area auto-scaling
/kind enhancement
/platform gcp

**What this PR does / why we need it**:
adds driver specific CSI label `topology.gke.io/zone` during machineDeployment creation

**Which issue(s) this PR fixes**:
Fixes #430

**Special notes for your reviewer**:

**Release note**:

```other operator
machineDeployment will have the label `topology.gke.io/zone` when created.
```

### CODE REVIEW

File: pkg/controller/worker/machines.go [hunk 1]
```diff
@@ -88,6 +92,9 @@ func (w *workerDelegate) generateMachineConfig(_ context.Context) error {
 		machineClasses     []map[string]interface{}
 		machineImages      []apisgcp.MachineImage
 	)
+	const (
+		csiMigrationVersion = "1.21"
```
Comment (kon-angelo) [inline pkg/controller/worker/machines.go:96]: Gardener has already dropped support for anything bellow 1.20 so the check is not necessary. Regardless the CSI migration version for gcp was 1.18 if I am not mistaken.

File: pkg/controller/worker/machines.go [hunk 2]
```diff
@@ -44,6 +46,8 @@ const (
 	maxGcpLabelCharactersSize = 63
 	// ResourceGPU is the GPU resource . It should be a non-negative integer.
 	ResourceGPU v1.ResourceName = "gpu"
+	// CSIDiskDriverTopologyKey is the label on persistent volumes that represents availability by zone.
+	CSIDiskDriverTopologyKey = "topology.gke.io/zone"
```
Comment (kon-angelo) [inline pkg/controller/worker/machines.go:50]: Out of convention, I would probably move this key to https://github.com/gardener/gardener-extension-provider-gcp/blob/master/pkg/gcp/types.go

**Review by kon-angelo — CHANGES_REQUESTED** [review]
Thank you for the contribution. Small comments from my side

File: pkg/controller/worker/machines.go (hunk 1 above)
Comment (elankath) [reply pkg/controller/worker/machines.go:96]: Ok, was un-certain about this. Will remove the check for CSI. thanks.

File: pkg/controller/worker/machines.go [hunk 3]
```diff
@@ -168,6 +170,9 @@ func (w *workerDelegate) generateMachineConfig(_ context.Context) error {
 		gceInstanceLabels := getGceInstanceLabels(w.worker.Name, pool)
 		isLiveMigrationAllowed := true
 
+		if err != nil {
+			return err
+		}
```
Comment (kon-angelo) [inline pkg/controller/worker/machines.go:175]: Is this still necessary ?

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Comment by elankath** [comment]
thanks Konstantinos.

**Pull request merged**